Koha/REST/V1/Auth.pm

```diff
@@ -110,6 +116,44 @@ sub authenticate_api_request {
 
     my $spec = $c->match->endpoint->pattern->defaults->{'openapi.op_spec'};
     my $authorization = $spec->{'x-koha-authorization'};
+
+    my $authorization_header = $c->req->headers->authorization;
+    if ($authorization_header and $authorization_header =~ /^Bearer /) {
+        my $server = Net::OAuth2::AuthorizationServer->new;
+        my $grant = $server->client_credentials_grant(Koha::OAuth::config);
+        my ($type, $token) = split / /, $authorization_header;
+        my ($valid_token, $error) = $grant->verify_access_token(
+            access_token => $token,
+        );
+
+        if ($valid_token) {
+            my $patron_id   = Koha::ApiKeys->find( $valid_token->{client_id} )->patron_id;
```

Comment:
This is the main issue I have here.. that we're automatically all permissions a user has on the system with the access_token we're creating.. there should really be a further table defining which permissions the user has given when creating the access token. (It's also why JWT is nice because you don't need to store it in a table at all.. the 'scopes' (which in essence really are just 'permissions') can be stored in the access token itself.

Reply:
I agree we could be using JWT (I'm not the OAuth2 server implementation author) but I also know we can phase into using them later too.